src/internal/internal_geadd.cc

```diff
@@ -376,10 +391,12 @@ void add(internal::TargetType<Target::Devices>,
             for (int64_t i = 0; i < B.mt(); ++i) {
                 for (int64_t j = 0; j < B.nt(); ++j) {
                     if (B.tileIsLocal(i, j) && device == B.tileDevice(i, j)) {
-                        // erase tmp local and remote device tiles;
-                        A.tileRelease(i, j, device);
-                        // decrement life for remote tiles
-                        A.tileTick(i, j);
+                        if (call_tile_tick) {
```

Comment:
If we put the conditional around the loops, it should slightly improve performance.  Especially when increasing the number of processes.  For a `2000x2000` grid of tiles, this type of loop takes about 0.3s for our normal 2d block cyclic lambdas.

Likewise for `tzadd<devices>`.